src/avalanchemq/queue/queue.cr

```diff
@@ -886,12 +886,20 @@ module AvalancheMQ
     def fsync_ack
     end
 
-    def to_json(builder : JSON::Builder)
+    def to_json(builder : JSON::Builder, limit : Int32 = -1)
       builder.object do
         details_tuple.each do |k, v|
           builder.field(k, v) unless v.nil?
         end
-        builder.field("consumer_details", @consumers)
+        builder.field("consumer_details") do
+          builder.array do
+            @consumers.each do |c|
```

Comment:
I think you could write `.each.take(limit)`

Reply:
`each` on ConsumerStore doesn't return an Iterable (returns Nil), it only yields each item inside each group.